Style not preserved with MergeYaml when values are written in flow style

## What version of OpenRewrite are you using?

I am using

- OpenRewrite 8.44.1
## What is the smallest, simplest way to reproduce the problem?

In order to reproduce the problem, I created this minimalist test that fails :


```java
package foo.bar;

import org.junit.jupiter.api.Test;
import org.openrewrite.test.RewriteTest;
import org.openrewrite.yaml.MergeYaml;

import static org.openrewrite.yaml.Assertions.yaml;

class MergeYamlFlowStyleTest implements RewriteTest {
    @Test
    void testYamlFlowStyle() {
        rewriteRun(spec -> spec
                        .recipe(new MergeYaml(//language=jsonpath
                                "$.jobs[?(@.name=='test')].plan[?(@.task=='sonar')]",
                                // language=yaml
                                "vars: { version: 10.3 }",
                                false,
                                null,
                                null
                        )),

                yaml(// language=yaml
                        """
                                jobs:
                                - name: test
                                  plan:
                                  - task: sonar
                                """,
                        // language=yaml
                        """
                                jobs:
                                - name: test
                                  plan:
                                  - task: sonar
                                    vars: { version: 10.3 }
                                """
                ));
    }
}

```

## What did you expect to see?
<!-- A code snippet, or a description of the behavior you expected helps us write a test to ensure the issue is fixed. -->
```yaml
jobs:
- name: test
  plan:
  - task: sonar
    vars: { version: 10.3 }
```

## What did you see instead?
<!-- A code snippet, or a description of the behavior you saw instead of the above expected result. -->
```yaml
jobs:
- name: test
  plan:
  - task: sonar
    vars: {
      version: 10.3 }
```

## What is the full stack trace of any errors you encountered?
<!-- When errors occur, please include the output of `--stacktrace` for Gradle or `--debug` for Maven. -->
``` diff
[ERROR] Failures: 
[ERROR]   MergeYamlFlowStyleTest.testYamlFlowStyle:12 [Unexpected result in "file.yaml":
diff --git a/file.yaml b/file.yaml
index d20dfbb..d455fe4 100644
--- a/file.yaml
+++ b/file.yaml
@@ -2,4 +2,5 @@ 
 - name: test
   plan:
   - task: sonar
-    vars: { version: 10.3 }
\ No newline at end of file
+    vars: {
+      version: 10.3 }
\ No newline at end of file
] 
expected: 
  "jobs:
  - name: test
    plan:
    - task: sonar
      vars: { version: 10.3 }"
 but was: 
  "jobs:
  - name: test
    plan:
    - task: sonar
      vars: {
        version: 10.3 }"
```

## Are you interested in [contributing a fix to OpenRewrite](https://github.com/openrewrite/.github/blob/main/CONTRIBUTING.md#contributing-fixes)?
<!-- Indicate if this is something you would like to work on, and how we can best support you in doing so. -->

--- a/rewrite-yaml/src/main/java/org/openrewrite/yaml/format/MinimumViableSpacingVisitor.java
+++ b/rewrite-yaml/src/main/java/org/openrewrite/yaml/format/MinimumViableSpacingVisitor.java
@@ -44,6 +44,10 @@ public class MinimumViableSpacingVisitor<P> extends YamlIsoVisitor<P> {
                     return e;
                 }
             }
+
+        if (" ".equals(e.getPrefix())) {
+            return e;
+        }
             return e.withPrefix("\n");
         }
         return e;

--- a/rewrite-yaml/src/main/java/org/openrewrite/yaml/format/MinimumViableSpacingVisitor.java
+++ b/rewrite-yaml/src/main/java/org/openrewrite/yaml/format/MinimumViableSpacingVisitor.java
@@ -45,9 +45,9 @@ public class MinimumViableSpacingVisitor<P> extends YamlIsoVisitor<P> {
                 }
             }
 
-        if (" ".equals(e.getPrefix())) {
-            return e;
-        }
+            if (" ".equals(e.getPrefix())) {
+                return e;
+            }
             return e.withPrefix("\n");
         }
         return e;

--- a/rewrite-yaml/src/test/java/org/openrewrite/yaml/MergeYamlTest.java
+++ b/rewrite-yaml/src/test/java/org/openrewrite/yaml/MergeYamlTest.java
@@ -1688,4 +1688,101 @@ class MergeYamlTest implements RewriteTest {
               """)
         );
     }
+
+    @Issue("https://github.com/openrewrite/rewrite/issues/4958")
+    @Test
+    void preserveSpacingWhenMergingFlowStyle() {
+        rewriteRun(spec -> spec
+            .recipe(new MergeYaml(//language=jsonpath
+              "$.jobs[?(@.name=='test')].plan[?(@.task=='sonar')]",
+              // language=yaml
+              "vars: { version: 10.3 }",
+              false,
+              null,
+              null
+            )),
+
+          yaml(// language=yaml
+            """
+              jobs:
+              - name: test
+                plan:
+                - task: sonar
+              """,
+            // language=yaml
+            """
+              jobs:
+              - name: test
+                plan:
+                - task: sonar
+                  vars: { version: 10.3 }
+              """
+          ));
+    }
+
+    @Issue("https://github.com/openrewrite/rewrite/issues/4958")
+    @Test
+    void preserveSpacingWhenMergingFlowStyleAttempt2() {
+        rewriteRun(spec -> spec
+            .recipe(new MergeYaml(//language=jsonpath
+              "$.jobs[?(@.name=='test')].plan[?(@.task=='sonar')]",
+              // language=yaml
+              "vars: { mapping: { version: 10.3 } }",
+              false,
+              null,
+              null
+            )),
+
+          yaml(// language=yaml
+            """
+              jobs:
+              - name: test
+                plan:
+                - task: sonar
+              """,
+            // language=yaml
+            """
+              jobs:
+              - name: test
+                plan:
+                - task: sonar
+                  vars: { mapping: { version: 10.3 } }
+              """
+          ));
+    }
+
+    @Issue("https://github.com/openrewrite/rewrite/issues/4958")
+    @Test
+    void preserveSpacingWhenMergingFlowStyleAttempt3() {
+        rewriteRun(spec -> spec
+            .recipe(new MergeYaml(//language=jsonpath
+              "$.jobs[?(@.name=='test')].plan[?(@.task=='sonar')]",
+              // language=yaml
+              """
+                vars: {
+                  version: 10.3 }
+                """,
+              false,
+              null,
+              null
+            )),
+
+          yaml(// language=yaml
+            """
+              jobs:
+              - name: test
+                plan:
+                - task: sonar
+              """,
+            // language=yaml
+            """
+              jobs:
+              - name: test
+                plan:
+                - task: sonar
+                  vars: {
+                    version: 10.3 }
+              """
+          ));
+    }
 }

--- a/rewrite-yaml/src/test/java/org/openrewrite/yaml/MergeYamlTest.java
+++ b/rewrite-yaml/src/test/java/org/openrewrite/yaml/MergeYamlTest.java
@@ -1685,7 +1685,8 @@ class MergeYamlTest implements RewriteTest {
                   script: >-
                     #!/bin/bash
                     echo "hello"
-              """)
+              """
+          )
         );
     }
 
@@ -1717,12 +1718,13 @@ class MergeYamlTest implements RewriteTest {
                 - task: sonar
                   vars: { version: 10.3 }
               """
-          ));
+          )
+        );
     }
 
     @Issue("https://github.com/openrewrite/rewrite/issues/4958")
     @Test
-    void preserveSpacingWhenMergingFlowStyleAttempt2() {
+    void preserveSpacingWhenMergingFlowStyleNested() {
         rewriteRun(spec -> spec
             .recipe(new MergeYaml(//language=jsonpath
               "$.jobs[?(@.name=='test')].plan[?(@.task=='sonar')]",
@@ -1748,12 +1750,13 @@ class MergeYamlTest implements RewriteTest {
                 - task: sonar
                   vars: { mapping: { version: 10.3 } }
               """
-          ));
+          )
+        );
     }
 
     @Issue("https://github.com/openrewrite/rewrite/issues/4958")
     @Test
-    void preserveSpacingWhenMergingFlowStyleAttempt3() {
+    void preserveSpacingWhenMergingFlowStyleWithNewline() {
         rewriteRun(spec -> spec
             .recipe(new MergeYaml(//language=jsonpath
               "$.jobs[?(@.name=='test')].plan[?(@.task=='sonar')]",
@@ -1783,6 +1786,7 @@ class MergeYamlTest implements RewriteTest {
                   vars: {
                     version: 10.3 }
               """
-          ));
+          )
+        );
     }
 }

